jsx/FilterForm.js

```diff
@@ -127,7 +127,7 @@ class FilterForm extends React.Component {
     }
     if (key) {
       filter[key] = {};
-      if (value.length === 0) {
+      if ((typeof (value) === 'string' || typeof (value) === 'object') && value.length === 0) {
```

Comment:
I don't think this will work either. I would replace 130-134 with
```
if (value) {     //all defined/non-null values must have some length, else empty string
  filter[key].value = Object.keys(value).length > 0 ? value : '';
} else {          //null and undefined handled here
  filter[key].value = '';
}
```

Reply:
Zain's edit works too